### PR TITLE
boost::format support for CLogger

### PR DESCRIPTION
**Pull request by alexvins**
We are tend to use boost::format for logging, so here is new log API extension to replace CLoggerStream

- also it is a first step to organize vstd namespace. New include directory introduced for such library headers.
- this PR includes changes in all build systems so I`m waiting for feedback at least about (un)successful compilation.

### CODE REVIEW

File: include/vstd/CLoggerBase.h [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+namespace vstd
+{
+	class DLL_LINKAGE CLoggerBase
+	{
+	public:
+		virtual ~CLoggerBase(){};
+
+		virtual void log(ELogLevel::ELogLevel level, const std::string & message) const = 0;
+
+		template<typename T, typename ... Args>
```
Comment (DjWarmonger) [inline include/vstd/CLoggerBase.h]: Won't all these templates significantly increase compiled code size?

Comment (alexvins) [inline include/vstd/CLoggerBase.h]: I don`t think so. Templates itself should be completely optimized out.

**Comment by ArseniyShestakov** [comment]
Just want to confirm this compile fine with Clang on Linux

**Comment by alexvins** [comment]
@DjWarmonger Any objections on merging this?

**Comment by DjWarmonger** [comment]
I need to find a time to check this. In my calendar ;)

**Comment by DjWarmonger** [comment]
This compiles just fine under MSVS. Ready for merge.

**Pull request merged**